include/swift/AST/Module.h

```diff
@@ -209,7 +210,20 @@ class ModuleDecl : public DeclContext, public TypeDecl {
 
 public:
   typedef ArrayRef<Located<Identifier>> AccessPathTy;
-  typedef std::pair<ModuleDecl::AccessPathTy, ModuleDecl*> ImportedModule;
+  
+  typedef struct _ImportedModule {
+    /// neither of these can be const because some things update after construction
+    ModuleDecl::AccessPathTy accessPathTy;
+    ModuleDecl * moduleDecl;
+  
+    /// should only be called to initialize when the module is loaded later
+    _ImportedModule() : accessPathTy(ModuleDecl::AccessPathTy()), moduleDecl(nullptr) {}
+    
+    /// normal constructor. will not accept a null moduleDecl pointer.
+    _ImportedModule(ModuleDecl::AccessPathTy accessPathTy, ModuleDecl* moduleDecl) : accessPathTy(accessPathTy), moduleDecl(moduleDecl) {
```

Comment:
Nitpick: This line, and a handful of other lines seem a bit too long. We usually try to stick to < 80 columns wherever possible.

Reply:
I think I got all of these, maybe not, as of af9a745.